tests/ml/test_ml_model_pytest.py

```diff
@@ -321,13 +321,27 @@ def test_decision_tree_regressor(self, compress_model_definition):
         es_model.delete_model()
 
     @requires_elasticsearch_version((8, 12))
-    @requires_sklearn
+    @requires_xgboost
     @pytest.mark.parametrize("compress_model_definition", [True, False])
-    def test_learning_to_rank(self, compress_model_definition):
-        # Train model
-        training_data = datasets.make_regression(n_features=2)
-        regressor = DecisionTreeRegressor()
-        regressor.fit(training_data[0], training_data[1])
+    @pytest.mark.parametrize(
+        "objective",
+        ["rank:ndcg", "rank:map", "rank:pairwise"],
+    )
+    def test_learning_to_rank(self, objective, compress_model_definition):
+        X, y = datasets.make_classification(
+            n_features=3, n_informative=2, n_redundant=1
+        )
+        rng = np.random.default_rng()
+        qid = rng.integers(0, 3, size=X.shape[0])
+
+        # Sort the inputs based on query index
+        sorted_idx = np.argsort(qid)
+        X = X[sorted_idx, :]
+        y = y[sorted_idx]
+        qid = qid[sorted_idx]
+
+        ranker = XGBRanker(objective=objective)
```

Comment:
With this change (switching to a ranker from a regressor) this test is now almost the same as `test_xgb_ranker()`, except it has a few additional assertions. Can we merge them or is there a reason for keeping both?

Reply:
`test_xgb_ranker` is doing testing for using a XGBRanker as a regressor. It is a test that I would like to keep since it allows to track regression in the inference process without worrying about others aspects from LTR (feature extractions, rescoring, ...)

In contrast `test_learning_to_rank` test the whole feature.